kivy/uix/video.py

```diff
@@ -31,21 +42,51 @@ def on_duration_change(instance, value):
 from kivy.core.video import Video as CoreVideo
 from kivy.resources import resource_find
 from kivy.properties import (BooleanProperty, NumericProperty, ObjectProperty,
-                             OptionProperty)
+                             OptionProperty, StringProperty)
+from kivy.logger import Logger
 
 
 class Video(Image):
     '''Video class. See module documentation for more information.
     '''
 
+    source = StringProperty(None)
+    '''Filename / source of your video.
+
+    :attr:`source` is a :class:`~kivy.properties.StringProperty` and
+    defaults to None.
+
+    .. versionchanged:: 2.1.0
+        The use of ``source`` property is deprecated. Use ``video_source``
```

Comment:
I don't think we should deprecate `source`. I can't see a real reason why using source as the default for both video and image should be removed eventually and break existing programs. Especially considering we won't be able to actually remove `source` because it's part of Image so at best we could say to not use it, but it's simpler to continue supporting it.

So, I'd rather we keep it officially supported, but rather in the docs point out that we have the two new properties that offer more control.

Reply:
> for both video and image

Using video path for image source just cause the image error log message and has no further advantage ;). But I'm fine with removing the deprecation warning if desired.

Reply:
Hmm, I assumed that the player would actually be able to get a preview image for some videos when a video is specified as source. Is that not the case? Did it never work in the past?

Reply:
Well, at least I've never seen what you describe. I also not see how this should integrate into the superclass on __init__, there's just the source property set and image tries to load an image texture as far as i can tell.

Reply:
ffpyplayer for example can load a gif as an image, I think. So I it may be able to similarly load a image from a video, but I'm not sure.

But either way, I guess even if it couldn't, I'd prefer to not deprecate. When a user sees that warning they'll hopefully look at the docs and use the new properties. If we could remove source altogether, I would probably feel differently, but since we can't I think as it is now is better.

Reply:
I already removed the deprecation warnings.